### PR TITLE
fix parallel execution

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -48,8 +48,8 @@ required = [
   branch = "master"
   name = "github.com/giantswarm/backoff"
 
-[[constraint]]
-  branch = "master"
+[[override]]
+  branch = "search-parallel"
   name = "github.com/giantswarm/certs"
```
Comment (xh3b4sd) [inline Gopkg.toml]: See https://github.com/giantswarm/certs/pull/36.

File: service/controller/v22/resource/s3bucket/create.go [hunk 2]
```diff
@@ -29,27 +29,29 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
...
+
 	for _, bucketInput := range createBucketsState {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating S3 bucket %#q", bucketInput.Name))
+		bucketInput := bucketInput
 
-		{
-			i := &s3.CreateBucketInput{
-				Bucket: aws.String(bucketInput.Name),
-			}
+		g.Go(func() error {
```
Comment (xh3b4sd) [inline service/controller/v22/resource/s3bucket/create.go]: Rewrote the parallel execution. Now the whole bucket setup is executed in series again but multiple buckets are executed in parallel. That was necessary due to creation issues.

File: service/controller/v22/resource/bridgezone/resource.go [hunk 3]
```diff
@@ -176,13 +176,8 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "getting intermediate zone ID")
 
 		id, err := r.findHostedZoneID(ctx, defaultGuest, intermediateZone)
-		if IsNotFound(err) {
-			// If the intermeidate zone is not found we are after
-			// the migraiton period and this resource becomes noop.
+		if err != nil {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "intermediate zone not found")
```
Comment (kopiczko) [inline service/controller/v22/resource/bridgezone/resource.go]: I think this log should stay in `if IsNotFound(err) {` block. There may be unrelated errors.

Comment (xh3b4sd) [reply service/controller/v22/resource/bridgezone/resource.go]: Then it gets redundant with the error handling, or we lose the information about the zone difference. Pick your evil.

Comment (kopiczko) [reply service/controller/v22/resource/bridgezone/resource.go]: Just to be clear code below is what I meant. I don't think you lose any information with that.

```go
		if IsNotFound(err) {
			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource reconciliation for custom object")
			return microerror.Mask(err)
		} else if err != nil {
			return microerror.Mask(err)
		}
```

Comment (xh3b4sd) [reply service/controller/v22/resource/bridgezone/resource.go]: The cancelation log was there before and it does not make sense to be there now IMO. Neither does the error checking. It is done below in one place. What is wrong with that? I can remove the log stating the zone is not found.

Comment (xh3b4sd) [reply service/controller/v22/resource/bridgezone/resource.go]: We talked. Logging is improved.

**Comment by xh3b4sd** [comment]
Fishing for LGTM.

**Pull request merged**